roachtest/pg_regress: accept recent diffs

yuzefovich opened the pull request:
We now disable `create_table_with_schema_locked` (the default has recently changed to `true`).

Fixes: #148331.
Release note: None

DrewKimball commented on pkg/cmd/roachtest/testdata/pg_regress/rangefuncs.diffs:
```diff
@@ -423,8 +242,7 @@ diff -U3 --label=/mnt/data1/postgres/src/test/regress/expected/rangefuncs.out --
  (rngfuncid int, rngfuncsubid int, rngfuncname text);
-+ERROR:  unknown function: getrngfunc6()
-+HINT:  There is probably a typo in function name. Or the intention was to use a user-defined function in the view query, which is currently not supported.
++ERROR:  a column definition list is required for functions returning "record"
```
I opened https://github.com/cockroachdb/cockroach/issues/149706 to track this.

DrewKimball commented on pkg/cmd/roachtest/testdata/pg_regress/create_index.diffs:
```diff
@@ -2569,12 +2569,15 @@ diff -U3 --label=/mnt/data1/postgres/src/test/regress/expected/create_index.out
++ERROR:  this schema change is disallowed because table "syscol_table" is locked and this operation cannot automatically unlock the table
++DETAIL:  To unlock the table, execute `ALTER TABLE syscol_table SET (schema_locked = false);`
++After the schema change completes, we recommend setting it back to true with `ALTER TABLE syscol_table SET (schema_locked = true);`.
++HINT:  Locking the table improves changefeed performance; see https://www.cockroachlabs.com/docs/_version_/changefeed-best-practices.html#lock-the-schema-on-changefeed-watched-tables
```
Should we unset `create_table_with_schema_locked` for pg_regress?

yuzefovich replied:
Good idea. I tried this and looks like `pg_regress` gets stuck now in the middle of all tests. I'll take another closer look later this week.

yuzefovich committed:
roachtest/pg_regress: accept recent diffs

We now disable `create_table_with_schema_locked` (the default has
recently changed to `true`).

Release note: None

spilchen reviewed (approved):
RLS files LGTM. Thanks for letting me know.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/148546#-:-OV7gYSqCwHbjwNwiygM:b37pvl1)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @DrewKimball)



<!-- Sent from Reviewable.io -->

yuzefovich reviewed (commented):
I updated the PR to no longer skip `identity` file given that #150021 has been fixed.

TFTRs!

bors r+

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/148546#-:-OVF-_l07sSLqkkbGFvx:b-d1ds3n)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @DrewKimball)



<!-- Sent from Reviewable.io -->

yuzefovich commented:
bors r+
